deployment/local_deployment.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: page
+title: Local Deployment
+---
+
+# Local or Network Deployment
+
+This page will give some instructions on how to run the OCS applications in a production environment on one or more servers connected through a local network. If you are interested in a persistant, high availability cloud deployment procedure, check out the [Cloud Deployment]({% link deployment/cloud_deployment.md %}) section.
+
+## Database Setup
+
+Most of the OCS application depend on a postgresQL database (with the postGIS plugin installed for the Science Archive). You can share the same postgres instance with all the applications, they will just each need their own database created, but they could share credentials for simplicity. Your options for running postgres are:
+
+* Use an existing postgres instance on your local network
+* Install postgres locally on your machine
+* Run a postgres/postgis docker container
+
+To start a postgresQL instance with postgis on a machine with docker forwarding it's port 5432 for local usage, you can use:
+
+```bash
+docker run --name ocs-postgis -p 0.0.0.0:5432:5432 -e POSTGRES_PASSWORD=postgres -d postgis/postgis
+```
+
+{% include notification.html message="By default, postgresQL within a docker container will store it's data in the docker containers filesystem. If you want to persist the database, should should either create a docker volume or volume mount the postgres data directory in the container to a folder on your local filesystem when you start the container." %}
+
+After you have a postgres instance running that you can connect to with a username/password (Assuming postgres/postgres here, change as needed), create a db for each OCS application you will use:
+
+```bash
+createdb --host 0.0.0.0 -p5432 -Upostgres -W my_db_name
```

Comment:
no real reason, I think they work interchangeably. Do you have a preference?

Reply:
My preference would be when starting up a web server have it listen on `0.0.0.0` so that it handles traffic sent to any of the addresses that the host has defined for itself, but when setting a url to point to a service or the db or whatever, use a single address like `127.0.0.1`. I came to that conclusion after reading [this page](https://www.howtogeek.com/225487/what-is-the-difference-between-127.0.0.1-and-0.0.0.0/#:~:text=0.0%3F-,127.0.,particular%20address'%20place%20holder) which defines `0.0.0.0` as "a non-routable meta-address used to designate an invalid, unknown, or non-applicable target (a ‘no particular address’ place holder)" and requests to it will be sent to the default route... not sure exactly how the default route is defined, but it seems safer to explicitly say the db is located at and should be accessed using `127.0.0.1` instead of `0.0.0.0` in case the default route resolves to something else.